engine/gfx/mon_icons.asm

```diff
@@ -129,11 +130,52 @@ SetOWFlyMonColor:
 	push bc
 	push af
 	call _GetFlyMonColor
-	ld de, wOBPals1
 	ld [wNeededPalIndex], a
+	ld b, a
+	push bc
+	ld b, 0
+	ld a, [wUsedObjectPals]
+	inc a
+	jr z, .unset_bit_found
+	dec a
+	ld b, -1
+.bit_check_loop
+	inc b
+	rrca
+	jr c, .bit_check_loop
+.unset_bit_found
+	ld a, b
+	pop bc
+	ld c, a
+	ld a, b
+	ld b, 0
+	ld hl, wLoadedObjPal0
+	add hl, bc
+	ld [hl], a
+	push bc
+	ld a, c
+	ld bc, 1 palettes
+	ld hl, wOBPals1
+	rst AddNTimes
+	ld d, h
+	ld e, l
 	farcall CopySpritePal
-	xor a ; OBJ 0
-	jr _SetMonColor
+	pop bc
+	ldh a, [hUsedOAMIndex]
+	cp ((NUM_SPRITE_OAM_STRUCTS - 1) * SPRITEOAMSTRUCT_LENGTH) - (13 * SPRITEOAMSTRUCT_LENGTH)
+	; if we didn't have enough OAM slots, we need to use the last 13 slots
+	ld a, (NUM_SPRITE_OAM_STRUCTS * SPRITEOAMSTRUCT_LENGTH) - (13 * SPRITEOAMSTRUCT_LENGTH)
+	jr nc, .got_oam_addr
+	ldh a, [hUsedOAMIndex]
+	; a = (NUM_SPRITE_OAM_STRUCTS - 13) * SPRITEOAMSTRUCT_LENGTH + 1
+	cpl
+	add (NUM_SPRITE_OAM_STRUCTS - 13) * SPRITEOAMSTRUCT_LENGTH + 1
+.got_oam_addr
```

Comment:
Okay, that answers that, the 12s/13s in weather.asm should be a constant in weather_constants.asm. Or gfx_constants.asm.
```suggestion
	ldh a, [hUsedOAMIndex]
	cp (NUM_SPRITE_OAM_STRUCTS - 13 - 1) * SPRITEOAMSTRUCT_LENGTH
	; if we didn't have enough OAM slots, we need to use the last 13 slots
	ld a, (NUM_SPRITE_OAM_STRUCTS - 13) * SPRITEOAMSTRUCT_LENGTH
	jr nc, .got_oam_addr
	ldh a, [hUsedOAMIndex]
	; a = (NUM_SPRITE_OAM_STRUCTS - 13) * SPRITEOAMSTRUCT_LENGTH + 1
	cpl
	add (NUM_SPRITE_OAM_STRUCTS - 13) * SPRITEOAMSTRUCT_LENGTH + 1
.got_oam_addr
```